spyder/plugins/projects/plugin.py

```diff
@@ -326,27 +326,15 @@ def update_project_actions(self):
     def create_new_project(self):
         """Create new project."""
         self.unmaximize()
-        active_project = self.current_active_project
         dlg = ProjectDialog(self, project_types=self.get_project_types())
         result = dlg.exec_()
         data = dlg.project_data
         root_path = data.get("root_path", None)
         project_type = data.get("project_type", EmptyProject.ID)
 
         if result:
-            # A project was not open before
-            if active_project is None:
-                if self.get_option('visible_if_project_open'):
-                    self.show_explorer()

```

Comment:
These lines were removed because they are redundant with code in the `open_project` method.
`create_new_project` checks the visibility of the project explorer and emits the closed project signal; then calls `_create_project` which calls `open_project`. `open_project` checks the visibility of the project explorer and emits the closed project signal. These only need to be done once, inside `open_project`. I see no possible scenario in which `create_new_project` is called and these actions should be taken, but `open_project` is not called.